nanoFramework.Aws.IoTCore.Devices/AwsSignatureVersion4/nfArrayListExtensions.cs

```diff
@@ -0,0 +1,137 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+namespace System.Collections
+{
+    /// <summary>
+    /// Extension methods used for canonicalization.
+    /// </summary>
+    public static class nfArrayListExtensions
+    {
+        /// <summary>
+        /// Sort an ArrayList
+        /// </summary>
+        /// <param name="items">Items in the ArrayList</param>
+        /// <param name="comparer">The Comparer</param>
+        public static void Sort(this ArrayList items, IComparer comparer)
```

Comment:
now I get your question on Discord :-)

Why not using a quick-sort algorithm? It's fast and simple: https://www.geeksforgeeks.org/quick-sort/
You even have it implemented directly in C# there.

Reply:
Thanks, I will look into that as a later optimization. Given it is currently working as expected (in the context of my pesudo tests matching full fat .Net), I would rather not mess with it until appropriate unit tests are added and fully end to end tested.